src/reanimated2/hook/useAnimatedStyle.ts

```diff
@@ -510,19 +511,25 @@ For more, see the docs: \`https://docs.swmansion.com/react-native-reanimated/doc
     return () => {
       stopMapper(mapperId);
     };
+    // eslint-disable-next-line react-hooks/exhaustive-deps
   }, dependencies);
 
   useEffect(() => {
-    animationsActive.value = true;
+    areAnimationsActive.value = true;
     return () => {
-      animationsActive.value = false;
+      areAnimationsActive.value = false;
     };
-  }, []);
+  }, [areAnimationsActive]);
```

Comment:
Did you tested this change? It could potentially break animation during render of component? Could you check what will happen if you trigger rerender on component with running animation?

Reply:
Tested it and it seems to be fine.